Non-Unity Compilation Failures

### Environment

How do you use Sentry?
Sentry.io

Which version of the SDK?
Latest from Github

How did you install the package? (Git-URL, Assetstore)
Git

Which version of Unreal?
5.0.0

Is this happening in Unreal (editor) or on a player like Android, iOS, Windows?
Compilation of the Editor (Non-Unity)
### Steps to Reproduce

1. Compile the Editor in Non-Unity mode 


### Expected Result
No compile failures


### Actual Result

````
In step 'Win64 Editor (Non Unity)':
Error 1/44
SentrySettingsCustomization.h(26): error C2065: 'IPropertyHandle': undeclared identifier
Error 2/44
SentrySettingsCustomization.h(26): error C2923: 'TSharedPtr': 'IPropertyHandle' is not a valid template type argument for parameter 'ObjectType'
SentrySettingsCustomization.h(26): note: see declaration of 'IPropertyHandle'
Error 3/44
SentrySettingsCustomization.h(26): error C2976: 'TSharedPtr': too few template arguments
SharedPointer.h(661): note: see declaration of 'TSharedPtr'
Error 4/44
SentrySettingsCustomization.h(26): error C2955: 'TSharedPtr': use of class template requires template argument list
SharedPointer.h(661): note: see declaration of 'TSharedPtr'
Error 5/44
SentrySettingsCustomization.h(27): error C2065: 'IPropertyHandle': undeclared identifier
Error 6/44
SentrySettingsCustomization.h(27): error C2923: 'TSharedPtr': 'IPropertyHandle' is not a valid template type argument for parameter 'ObjectType'
SentrySettingsCustomization.h(27): note: see declaration of 'IPropertyHandle'
Error 7/44
SentrySettingsCustomization.h(27): error C2976: 'TSharedPtr': too few template arguments
SharedPointer.h(661): note: see declaration of 'TSharedPtr'
Error 8/44
SentrySettingsCustomization.h(27): error C2955: 'TSharedPtr': use of class template requires template argument list
SharedPointer.h(661): note: see declaration of 'TSharedPtr'
Error 9/44
SentrySettingsCustomization.h(28): error C2065: 'IPropertyHandle': undeclared identifier
Error 10/44
SentrySettingsCustomization.h(28): error C2923: 'TSharedPtr': 'IPropertyHandle' is not a valid template type argument for parameter 'ObjectType'
SentrySettingsCustomization.h(28): note: see declaration of 'IPropertyHandle'
Error 11/44
SentrySettingsCustomization.h(28): error C2976: 'TSharedPtr': too few template arguments
SharedPointer.h(661): note: see declaration of 'TSharedPtr'
Error 12/44
SentrySettingsCustomization.h(28): error C2955: 'TSharedPtr': use of class template requires template argument list
SharedPointer.h(661): note: see declaration of 'TSharedPtr'
Error 13/44
SentryEditorModule.cpp(15): error C2065: 'FPropertyEditorModule': undeclared identifier
Error 14/44
SentryEditorModule.cpp(15): error C2065: 'PropertyModule': undeclared identifier
Error 15/44
SentryEditorModule.cpp(15): error C2672: 'FModuleManager::LoadModuleChecked': no matching overloaded function found
Error 16/44
SentryEditorModule.cpp(15): error C2974: 'FModuleManager::LoadModuleChecked': invalid template argument for 'TModuleInterface', type expected
ModuleManager.h(324): note: see declaration of 'FModuleManager::LoadModuleChecked'
Error 17/44
SentryEditorModule.cpp(16): error C2065: 'PropertyModule': undeclared identifier
Error 18/44
SentryEditorModule.cpp(16): error C2653: 'FOnGetDetailCustomizationInstance': is not a class or namespace name
Error 19/44
SentryEditorModule.cpp(16): error C3861: 'CreateStatic': identifier not found
Error 20/44
SentryEditorModule.cpp(17): error C2065: 'PropertyModule': undeclared identifier
Error 21/44
SentrySettingsCustomization.cpp(33): error C2065: 'SBorder': undeclared identifier
Error 22/44
SentrySettingsCustomization.cpp(33): error C2672: 'MakeTDecl': no matching overloaded function found
Error 23/44
SentrySettingsCustomization.cpp(33): error C2974: 'MakeTDecl': invalid template argument for 'WidgetType', type expected
DeclarativeSyntaxSupport.h(940): note: see declaration of 'MakeTDecl'
Error 24/44
SentrySettingsCustomization.cpp(33): error C2653: 'SBorder': is not a class or namespace name
Error 25/44
SentrySettingsCustomization.cpp(33): error C3861: 'FArguments': identifier not found
Error 26/44
SentrySettingsCustomization.cpp(54): error C2678: binary '=': no operator found which takes a left-hand operand of type 'TSharedPtr' (or there is no acceptable conversion)
SharedPointer.h(931): note: could be 'TSharedPtr<ObjectType,InMode> &TSharedPtr<ObjectType,InMode>::operator =(TSharedPtr<ObjectType,InMode> &&)'
SharedPointer.h(924): note: or       'TSharedPtr<ObjectType,InMode> &TSharedPtr<ObjectType,InMode>::operator =(const TSharedPtr<ObjectType,InMode> &)'
SharedPointer.h(911): note: or       'TSharedPtr<ObjectType,InMode> &TSharedPtr<ObjectType,InMode>::operator =(SharedPointerInternals::FNullTag *)'
SentrySettingsCustomization.cpp(54): note: while trying to match the argument list '(TSharedPtr, TSharedRef<IPropertyHandle,ESPMode::ThreadSafe>)'
Error 27/44
SentrySettingsCustomization.cpp(54): error C2582: 'operator =' function is unavailable in 'TSharedPtr'
Error 28/44
SentrySettingsCustomization.cpp(55): error C2678: binary '->': no operator found which takes a left-hand operand of type 'TSharedPtr' (or there is no acceptable conversion)
SharedPointer.h(1072): note: could be 'ObjectType *TSharedPtr<ObjectType,InMode>::operator ->(void) const'
SentrySettingsCustomization.cpp(55): note: while trying to match the argument list '(TSharedPtr)'
Error 29/44
SentrySettingsCustomization.cpp(55): error C2039: 'SetOnPropertyValueChanged': is not a member of 'TSharedPtr'
SharedPointer.h(661): note: see declaration of 'TSharedPtr'
Error 30/44
SentrySettingsCustomization.cpp(58): error C2678: binary '=': no operator found which takes a left-hand operand of type 'TSharedPtr' (or there is no acceptable conversion)
SharedPointer.h(931): note: could be 'TSharedPtr<ObjectType,InMode> &TSharedPtr<ObjectType,InMode>::operator =(TSharedPtr<ObjectType,InMode> &&)'
SharedPointer.h(924): note: or       'TSharedPtr<ObjectType,InMode> &TSharedPtr<ObjectType,InMode>::operator =(const TSharedPtr<ObjectType,InMode> &)'
SharedPointer.h(911): note: or       'TSharedPtr<ObjectType,InMode> &TSharedPtr<ObjectType,InMode>::operator =(SharedPointerInternals::FNullTag *)'
SentrySettingsCustomization.cpp(58): note: while trying to match the argument list '(TSharedPtr, TSharedRef<IPropertyHandle,ESPMode::ThreadSafe>)'
Error 31/44
SentrySettingsCustomization.cpp(58): error C2582: 'operator =' function is unavailable in 'TSharedPtr'
Error 32/44
SentrySettingsCustomization.cpp(59): error C2678: binary '->': no operator found which takes a left-hand operand of type 'TSharedPtr' (or there is no acceptable conversion)
SharedPointer.h(1072): note: could be 'ObjectType *TSharedPtr<ObjectType,InMode>::operator ->(void) const'
SentrySettingsCustomization.cpp(59): note: while trying to match the argument list '(TSharedPtr)'
Error 33/44
SentrySettingsCustomization.cpp(59): error C2039: 'SetOnPropertyValueChanged': is not a member of 'TSharedPtr'
SharedPointer.h(661): note: see declaration of 'TSharedPtr'
Error 34/44
SentrySettingsCustomization.cpp(62): error C2678: binary '=': no operator found which takes a left-hand operand of type 'TSharedPtr' (or there is no acceptable conversion)
SharedPointer.h(931): note: could be 'TSharedPtr<ObjectType,InMode> &TSharedPtr<ObjectType,InMode>::operator =(TSharedPtr<ObjectType,InMode> &&)'
SharedPointer.h(924): note: or       'TSharedPtr<ObjectType,InMode> &TSharedPtr<ObjectType,InMode>::operator =(const TSharedPtr<ObjectType,InMode> &)'
SharedPointer.h(911): note: or       'TSharedPtr<ObjectType,InMode> &TSharedPtr<ObjectType,InMode>::operator =(SharedPointerInternals::FNullTag *)'
SentrySettingsCustomization.cpp(62): note: while trying to match the argument list '(TSharedPtr, TSharedRef<IPropertyHandle,ESPMode::ThreadSafe>)'
Error 35/44
SentrySettingsCustomization.cpp(62): error C2582: 'operator =' function is unavailable in 'TSharedPtr'
Error 36/44
SentrySettingsCustomization.cpp(63): error C2678: binary '->': no operator found which takes a left-hand operand of type 'TSharedPtr' (or there is no acceptable conversion)
SharedPointer.h(1072): note: could be 'ObjectType *TSharedPtr<ObjectType,InMode>::operator ->(void) const'
SentrySettingsCustomization.cpp(63): note: while trying to match the argument list '(TSharedPtr)'
Error 37/44
SentrySettingsCustomization.cpp(63): error C2039: 'SetOnPropertyValueChanged': is not a member of 'TSharedPtr'
SharedPointer.h(661): note: see declaration of 'TSharedPtr'
Error 38/44
SentrySettingsCustomization.cpp(69): error C2678: binary '->': no operator found which takes a left-hand operand of type 'TSharedPtr' (or there is no acceptable conversion)
SharedPointer.h(1072): note: could be 'ObjectType *TSharedPtr<ObjectType,InMode>::operator ->(void) const'
SentrySettingsCustomization.cpp(69): note: while trying to match the argument list '(TSharedPtr)'
Error 39/44
SentrySettingsCustomization.cpp(69): error C2039: 'GetValue': is not a member of 'TSharedPtr'
SharedPointer.h(661): note: see declaration of 'TSharedPtr'
Error 40/44
SentrySettingsCustomization.cpp(77): error C2678: binary '->': no operator found which takes a left-hand operand of type 'TSharedPtr' (or there is no acceptable conversion)
SharedPointer.h(1072): note: could be 'ObjectType *TSharedPtr<ObjectType,InMode>::operator ->(void) const'
SentrySettingsCustomization.cpp(77): note: while trying to match the argument list '(TSharedPtr)'
Error 41/44
SentrySettingsCustomization.cpp(77): error C2039: 'GetValue': is not a member of 'TSharedPtr'
SharedPointer.h(661): note: see declaration of 'TSharedPtr'
Error 42/44
SentrySettingsCustomization.cpp(85): error C2678: binary '->': no operator found which takes a left-hand operand of type 'TSharedPtr' (or there is no acceptable conversion)
SharedPointer.h(1072): note: could be 'ObjectType *TSharedPtr<ObjectType,InMode>::operator ->(void) const'
SentrySettingsCustomization.cpp(85): note: while trying to match the argument list '(TSharedPtr)'
Error 43/44
SentrySettingsCustomization.cpp(85): error C2039: 'GetValue': is not a member of 'TSharedPtr'
SharedPointer.h(661): note: see declaration of 'TSharedPtr'
Error 44/44
SentrySettingsCustomization.cpp(94): error C2079: 'PropertiesFile' uses undefined class 'FConfigFile'
```

### Any logs or screenshots

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -9,6 +9,7 @@
 ### Fixes
 
 - Packaged plugin `EngineVersion` should include `.0` patch version ([#101](https://github.com/getsentry/sentry-unreal/pull/101))
+- Plugin packaging issues on Windows ([#110](https://github.com/getsentry/sentry-unreal/pull/110))
 
 ### Dependencies
 

--- a/README.md
+++ b/README.md
@@ -9,9 +9,34 @@
 Sentry SDK for Unreal Engine
 ===========
 
-Sentry [has a stable support for the Unreal Engine crash reporter](https://docs.sentry.io/platforms/unreal/setup-crashreport/).
-This SDK is a prototype, that packages Sentry's mobile SDKs into a Unreal Engine package.
+This project is an SDK for Unreal Engine that wraps different Sentry's SDKs for both desktop and mobile. Also, it [has a stable support for the Unreal Engine crash reporter](https://docs.sentry.io/platforms/unreal/setup-crashreport/).
 
+## Downloads
+
+The SDK can be downloaded from the [Releases] page, which also lists the
+changelog of every version.
+
+[releases]: https://github.com/getsentry/sentry-unreal/releases
+
+## Supported Platforms and Unreal Engine version
+
+The SDK currently supports and is tested on the following platforms:
+
+- Windows
+- macOS
+- Android
+- iOS
+
+The SDK complies with three latest engine versions.
+
+## Known Limitations
+
+- Currently automatic crash capturing works only for mobile platforms. In order to capture crahes on Windows and Mac
+[Crash Reporter has to be configured](https://docs.sentry.io/platforms/unreal/setup-crashreport/)
+
+## Development
+
+Please see the [contribution guide](./CONTRIBUTING.md).
 
 ## Resources
 

--- a/plugin-dev/Source/Sentry/Private/SentrySettings.cpp
+++ b/plugin-dev/Source/Sentry/Private/SentrySettings.cpp
@@ -3,6 +3,7 @@
 #include "SentrySettings.h"
 
 #include "Misc/Paths.h"
+#include "Misc/ConfigCacheIni.h"
 
 USentrySettings::USentrySettings(const FObjectInitializer& ObjectInitializer)
 	: Super(ObjectInitializer)

--- a/plugin-dev/Source/SentryEditor/Private/SentryEditorModule.cpp
+++ b/plugin-dev/Source/SentryEditor/Private/SentryEditorModule.cpp
@@ -5,6 +5,7 @@
 #include "SentrySettingsCustomization.h"
 
 #include "Modules/ModuleManager.h"
+#include "PropertyEditorModule.h"
 
 #define LOCTEXT_NAMESPACE "FSentryEditorModule"
 

--- a/plugin-dev/Source/SentryEditor/Private/SentrySettingsCustomization.cpp
+++ b/plugin-dev/Source/SentryEditor/Private/SentrySettingsCustomization.cpp
@@ -8,8 +8,11 @@
 #include "DetailWidgetRow.h"
 #include "EditorStyleSet.h"
 #include "Misc/Paths.h"
+#include "Misc/ConfigCacheIni.h"
+#include "PropertyHandle.h"
 
 #include "Widgets/Text/SRichTextBlock.h"
+#include "Widgets/Layout/SBorder.h"
 
 TSharedRef<IDetailCustomization> FSentrySettingsCustomization::MakeInstance()
 {

--- a/plugin-dev/Source/SentryEditor/Public/SentrySettingsCustomization.h
+++ b/plugin-dev/Source/SentryEditor/Public/SentrySettingsCustomization.h
@@ -4,6 +4,8 @@
 
 #include "IDetailCustomization.h"
 
+class IPropertyHandle;
+
 class FSentrySettingsCustomization : public IDetailCustomization
 {
 public:

--- a/plugin-dev/Source/SentryEditor/SentryEditor.Build.cs
+++ b/plugin-dev/Source/SentryEditor/SentryEditor.Build.cs
@@ -23,7 +23,8 @@ public class SentryEditor : ModuleRules
                 "Slate",
                 "SlateCore",
                 "EditorStyle",
-                "Sentry"
+                "Sentry",
+                "PropertyEditor"
             }
         );
     }

